### DISCUSSION TITLE
Script error, when adding a PnP modern search webpart.

### DISCUSSION THREAD

**Discussion by AdamSzabo0721** [microsoft-search/pnp-modern-search#3339] [Q&A] [answered]
Hi,

**Background:**
We are using the tenant of our mother company, so the IT can react really slowly unfortunately. After months we have finally got the PnP Modern search installed and made available for every site. Unfortunately, when we add any of the webparts we get an error straight away.

**The issue:**
Adding any PnP returns the following error message:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/151152140/d09e534e-edcf-43a2-b886-d20a25ebd07d)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/151152140/e9a730da-11b0-49e9-8ecd-e6389035700c)

**Analysis so far:**
We have tested,

- multiple browsers
- multiple users
- different sites and pages
- deleted browser cache and cookies
So the issue seems to be not on the end-user side.

I have searched for possibly similar discussion and scenarios here and online, but could not find an exact fit.
Our SP consultants could not find the issue, and raising a simple request to our IT without exact information or to do would take months to be checked on again.

Thanks in advance for every possible comment, advice!

**Comment by kasperbolarsen** [comment, marked as answer]
That error does look a lot like the one I encountered recently, see https://twitter.com/kasperbolarsen/status/1723095490084884741 . 
The root cause was the users ( in my case the users were external guests) couldn't access the tenant App Catalog. Would it be possible to verify this possible root cause?
Another option if Corp IT isn't playing ball could be to install the PnP Modern Search package in the Site Collection App Catalog?

**Reply by AdamSzabo0721** [reply]
Thank you for the suggestion! It seemed to be the issue indeed. Every user for who we provided access to the App Catalog now can use the webparts, and the users without it still receiving the error message. 

Do I understand correctly that I can request the app to be installed directly in a Site collections App Catalog so it would work on my Site collection? I do need to keep fight with Corp IT, since there is no global access for the App Catalog to our users currently. :/

**Reply by kasperbolarsen** [reply]
Using the Site collection app catalog looks like an option in your case. However the issue with no read access to the tenant app catalog is not limited to PnP Modern Search,  it is the case for all 3rd party SPFx web parts .

https://learn.microsoft.com/en-us/sharepoint/dev/general-development/site-collection-app-catalog

**Reply by AdamSzabo0721** [reply]
Understood. Thank you again for the quick and great help!
